core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4646,6 +4676,62 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
             analysis.addRowFilter(table, expression);
         }
 
+        private void analyzeCheckConstraint(Table table, QualifiedObjectName name, Scope scope, ViewExpression filter)
```

Comment:
Using `filter` for the argument name is a little misleading. Rename it to `constraint`